icu4c/source/test/intltest/tzfmttst.cpp

```diff
@@ -1302,4 +1303,721 @@ TimeZoneFormatTest::TestFormatTZDBNamesAllZoneCoverage(void) {
+        "Europe/Prague",
+        "Europe/Riga",
+        "Europe/Rome",
+        "Europe/Samara",
+        "Europe/San_Marino",
+        "Europe/Sarajevo",
+        "Europe/Simferopol",
+        "Europe/Skopje",
+        "Europe/Sofia",
+        "Europe/Tallinn",
+        "Europe/Tirane",
+        "Europe/Tiraspol",
+        "Europe/Ulyanovsk",
+        "Europe/Uzhgorod",
+        "Europe/Vaduz",
+        "Europe/Vienna",
+        "Europe/Vilnius",
+        "Europe/Volgograd",
+        "Europe/Warsaw",
+        "Europe/Zagreb",
+        "Europe/Zaporozhye",
+        "Europe/Zurich",
+        "Indian/Chagos",
+        "Indian/Cocos",
+        "Indian/Comoro",
+        "Indian/Kerguelen",
+        "Indian/Mahe",
+        "Indian/Maldives",
+        "Indian/Reunion",
+        "Mexico/BajaSur",
+        "Mexico/General",
+        "Pacific/Apia",
+        "Pacific/Auckland",
+        "Pacific/Bougainville",
+        "Pacific/Chuuk",
+        "Pacific/Enderbury",
+        "Pacific/Fakaofo",
+        "Pacific/Fiji",
+        "Pacific/Galapagos",
+        "Pacific/Gambier",
+        "Pacific/Guadalcanal",
+        "Pacific/Guam",
+        "Pacific/Honolulu",
+        "Pacific/Kosrae",
+        "Pacific/Kwajalein",
+        "Pacific/Majuro",
+        "Pacific/Marquesas",
+        "Pacific/Midway",
+        "Pacific/Nauru",
+        "Pacific/Niue",
+        "Pacific/Norfolk",
+        "Pacific/Noumea",
+        "Pacific/Pago_Pago",
+        "Pacific/Palau",
+        "Pacific/Pohnpei",
+        "Pacific/Ponape",
+        "Pacific/Saipan",
+        "Pacific/Samoa",
+        "Pacific/Tarawa",
+        "Pacific/Truk",
+        "Pacific/Wake",
+        "Pacific/Wallis",
+        "Pacific/Yap",
+        "US/Alaska",
+        "US/Arizona",
+        "US/Hawaii",
+        "US/Michigan",
+        "US/Pacific",
+        "US/Pacific-New",
+        "US/Samoa",
+    };
+
+    int32_t nTZ = static_cast<int32_t>(sizeof(tz) / sizeof(char*));
+    std::string locale;
+    for (int32_t tzidx = 0; tzidx < nTZ; tzidx++) {
+        locale = "en-u-tz-";
+        locale += tz[tzidx];
+        Locale l(locale.c_str());
+        UErrorCode status = U_ZERO_ERROR;
+        SimpleDateFormat fmt(UnicodeString("V"), l, status);
+        if (U_FAILURE(status)) {
+            dataerrln("Error creating SimpleDateFormat(%s) - %s", locale.c_str(), u_errorName(status));
+            continue;
+        }
+        UnicodeString result;
+        fmt.format(now, result, status);
+        if (U_FAILURE(status)) {
+            dataerrln("Error format with locale %s - %s", locale.c_str(), u_errorName(status));
+            continue;
+        }
+        std::string result_str;
+        result.toUTF8String(result_str);
+        if (result_str != tz[tzidx]) {
+            dataerrln("Error format('V') with locale %s - expected %s but got %s", locale.c_str(), tz[tzidx], result_str.c_str());
+        }
+    }
+
+    int32_t nTimezoen = static_cast<int32_t>(sizeof(timezone) / sizeof(char*));
```

Comment:
Please always use the ICU4C macro `UPRV_LENGTHOF()` from `cmemory.h` to calculate the length of static data structures.